doc/content/design/sm-supported-image-formats.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Add supported image formats in sm-list
+layout: default
+design_doc: true
+revision: 2
+status: proposed
+---
+
+# Introduction
+
+At XCP-ng, we are enhancing support for QCOW2 images in SMAPI. The primary
+motivation for this change is to overcome the 2TB size limitation imposed
+by the VHD format. By adding support for QCOW2, a Storage Repository (SR) will
+be able to host disks in VHD and/or QCOW2 formats, depending on the SR type.
+In the future, additional formats—such as VHDx—could also be supported.
+
+We need a mechanism to expose to end users which image formats are supported
+by a given SR. The proposal is to extend the SM API object with a new field
+that clients (such as XenCenter, XenOrchestra, etc.) can use to determine the
+available formats.
+
+# Design Proposal
+
+To expose the available image formats to clients (e.g., XenCenter, XenOrchestra, etc.),
+we propose adding a new field called `supported-image-formats` to the Storage Manager (SM)
+module. This field will be included in the output of the `SM.get_all_records` call.
+
+The `supported-image-formats` field will be populated by retrieving information
+from the SMAPI drivers. Specifically, each driver will update its `DRIVER_INFO`
+dictionary with a new key, `supported_image_formats`, which will contain a list
+of strings representing the supported image formats
+(for example: `["vhd", "raw", "qcow2"]`).
+
+The list designates the driver's preferred VDI format as its first entry. That
+means that when migrating a VDI, the destination storage repository will
+attempt to create a VDI in this preferred format. If the default format cannot
```

Comment:
Is the intention to automatically change the format of a VDI to the "preferred" format when doing a storage migration? Is there no choice or desire to keep the current format of the VDI?

Reply:
currently the "preferred" format is imposed by the storage driver. So we cannot change it. But yes in the next release it makes sense to at least be able to either keep the same format or use the qcow2 format. 
So the current design does not take this into account but I think that effectively adding two fields for the destination format during the migration as well as a field for the format used when creating the VDI seems important finally.

Reply:
It's just because of how migration work at the moment, creating a new VDI (with the default preferred format) and streaming the data of the old VDI in the new one.